### DISCUSSION TITLE
How to Set Up ARM64 Code Server from tar.gz Release?

### DISCUSSION THREAD

**Discussion by Future-Outlier** [coder/code-server#6555] [Q&A] [answered]
## Description
I'm encountering an issue while trying to set up Code Server on an ARM64 system, following the instructions from the [official documentation](https://coder.com/docs/code-server/latest/install#standalone-releases).
But I found that amd64 works, but arm64 doesn't.
It sounds that the binary file of node is wrong.

## Setup Process
```bash
wget --no-check-certificate -O /tmp/code-server/code-server-4.19.0-linux-arm64.tar.gz https://github.com/coder/code-server/releases/download/v4.19.0/code-server-4.19.0-linux-arm64.tar.gz
tar -xzf ./code-server-4.19.0-linux-arm64.tar.gz -C ./
cd code-server-4.19.0-linux-arm64
./bin/code-server
```

## In ARM64
![image](https://github.com/coder/code-server/assets/76461262/50108aa4-8e71-4e16-93fe-dd26d8b44d35)
## In AMD64
![image](https://github.com/coder/code-server/assets/76461262/db25e498-7b48-48f1-a9eb-948e567b0b47)

**Comment by code-asher** [comment]
It is possible the build script has an bug.  We try to download Node straight from nodejs.org for the appropriate architecture.

https://github.com/coder/code-server/blob/0c98611e6b43803a9d5dba222d7023b569abfb49/.github/workflows/release.yaml#L174-L179

If you download and extract https://nodejs.org/dist/v18.15.0/node-v18.15.0-linux-arm64.tar.xz does the Node binary in there work?

**Reply by Future-Outlier** [reply]
No, it doesn't work
![image](https://github.com/coder/code-server/assets/76461262/476419fd-0bb6-4c8e-a0b8-5623a89b7fb8)
I think we need to change the script and point to the right version right?

**Reply by code-asher** [reply, marked as answer]
Very interesting!  Thank you for confirming.

If this is an arm64 system `node-v18.15.0-linux-arm64` is surely the right version.  Does `uname -a` report the right architecture?  If the amd64 binary works this is probably an amd64 system.

**Reply by Future-Outlier** [reply]
Thank you really much, let me try to build an arm64 image and tell you my result!
